valohai/internals/yaml.py

```diff
@@ -40,8 +40,13 @@ def generate_step(
     for key, value in inputs.items():
         has_wildcards = any("*" in uri for uri in value)
         keep_directories = KeepDirectories.SUFFIX.value if has_wildcards else False
+        empty_default = not value or len(value) == 0 or len(value) == 1 and not value[0]
```

Comment:
If `len(value)` is for `""`-empty string and we want the checks to be same for `value` and `value[0],` then the case where `value[0] = ""` should be handled? `len(value[0]) == ""`

Reply:
Actually we always expect a list here and not a string.

So we check for None, we check for empty list and for an non-empty list we check the first item. Having multiple empty items is such an exotic thing here, that we don't care.

Reply:
Hmm looking at the InputDict type it is a string. wtf. Good catch!